### PR TITLE
 add working tables-view with time left as duration

### PR DESCRIPTION
**Pull request by BertPie**

### CODE REVIEW

File: ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dto/TableTimeLeft.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.infoshareacademy.niewiem.pojo.Table;
+
+import java.time.Duration;
+
+public class TableTimeLeft {
+    private Table table;
+    private Duration timeLeft;
+
+    public TableTimeLeft() {
```
Comment (isacademyci) [inline ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dto/TableTimeLeft.java]: CRITICAL Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1186)

File: ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dao/ReservationDao.java [hunk 2]
```diff
@@ -14,12 +18,17 @@
     @PersistenceContext
     private EntityManager entityManager;
 
+    @Inject
+    private TableDao tableDao;
+
     public Long save(Reservation reservation){
+        // todo: check if reservation has no conflict
```
Comment (isacademyci) [inline ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dao/ReservationDao.java]: INFO Complete the task associated to this TODO comment. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1135)

File: ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dao/ReservationDao.java [hunk 3]
```diff
@@ -14,12 +18,17 @@
...
         entityManager.persist(reservation);
         return reservation.getId();
     }
 
     public Reservation update (Reservation reservation){
+        // todo: check if reservation has no conflict
```
Comment (isacademyci) [inline ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/dao/ReservationDao.java]: INFO Complete the task associated to this TODO comment. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1135)

**Comment by isacademyci** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Hall.java#L61](https://github.com/infoshareacademy/jjdd5-niewiem/blob/85d33f13e0c851476dc8a94b5a7a096e460276ba/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Hall.java#L61): Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [Table.java#L104](https://github.com/infoshareacademy/jjdd5-niewiem/blob/85d33f13e0c851476dc8a94b5a7a096e460276ba/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L104): Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1149)
1. ![MINOR][MINOR] [Table.java#L40](https://github.com/infoshareacademy/jjdd5-niewiem/blob/85d33f13e0c851476dc8a94b5a7a096e460276ba/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L40): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [Table.java#L90](https://github.com/infoshareacademy/jjdd5-niewiem/blob/85d33f13e0c851476dc8a94b5a7a096e460276ba/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L90): This class overrides "equals()" and should therefore also override "hashCode()". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1206)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by isacademyci** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Hall.java#L61](https://github.com/infoshareacademy/jjdd5-niewiem/blob/d8c13d0371329cfd33199e8c64a4fff55d11ddc2/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Hall.java#L61): Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1149)
1. ![MAJOR][MAJOR] [Table.java#L104](https://github.com/infoshareacademy/jjdd5-niewiem/blob/d8c13d0371329cfd33199e8c64a4fff55d11ddc2/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L104): Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1149)
1. ![MINOR][MINOR] [Table.java#L40](https://github.com/infoshareacademy/jjdd5-niewiem/blob/d8c13d0371329cfd33199e8c64a4fff55d11ddc2/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L40): Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS00117)
1. ![MINOR][MINOR] [Table.java#L90](https://github.com/infoshareacademy/jjdd5-niewiem/blob/d8c13d0371329cfd33199e8c64a4fff55d11ddc2/ClubManagerApp/src/main/java/com/infoshareacademy/niewiem/pojo/Table.java#L90): This class overrides "equals()" and should therefore also override "hashCode()". [rule](http://localhost:9000/coding_rules#rule_key=squid%3AS1206)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/freemarker/TemplateProvider.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 @ApplicationScoped
 public class TemplateProvider {
-    public static final String LAYOUT_NAME = "layout/app";
+    public static final String LAYOUT_NAME = "layouts/app";
```
Comment (MichalNowakowski) [inline ClubManagerWeb/src/main/java/com/infoshareacademy/niewiem/freemarker/TemplateProvider.java]: zaimportuj LAYOUT_NAME z TemplateProvider (Michal dodal do developa)

**Pull request merged**